src/twisted/web/server.py

```diff
@@ -500,6 +505,27 @@ def getRootURL(self):
         return self.appRootURL
 
 
+    def _handleStar(self):
+        """
+        Handle receiving a request whose path is '*'.
+
+        RFC 7231 defines an OPTIONS * request as being something that a client
+        can send as a low-effort way to probe server capabilities or readiness.
+        Rather than bother the user with this, we simply fast-path it back to
+        an empty 200 OK. Any non-OPTIONS verb gets a 405 Method Not Allowed
+        telling the client they can only use OPTIONS.
+        """
+        if self.method == b'OPTIONS':
+            self.setResponseCode(http.OK)
+        else:
+            self.setResponseCode(http.NOT_ALLOWED)
+            self.setHeader(b'Allow', b'OPTIONS')
+
```

Comment:
You sure this is right?

I suspect that `*` is not a valid resource name (outside of an `OPTIONS` request), in which case `BAD_REQUEST` seems more appropriate.  If it's a valid resource name, then the server should get a crack at handling the request and not get short-circuited, no?